internal/envoy/v3/bootstrap.go

```diff
@@ -251,6 +253,51 @@ func bootstrapConfig(c *envoy.BootstrapConfig) *envoy_bootstrap_v3.Bootstrap {
 			Address:   UnixSocketAddress(c.GetAdminAddress(), c.GetAdminPort()),
 		},
 	}
+	if c.MaximumHeapSizeBytes > 0 {
+		bootstrap.OverloadManager = &envoy_config_overload_v3.OverloadManager{
+			RefreshInterval: protobuf.Duration(250 * time.Millisecond),
+			ResourceMonitors: []*envoy_config_overload_v3.ResourceMonitor{
+				{
+					Name: "envoy.resource_monitors.fixed_heap",
+					ConfigType: &envoy_config_overload_v3.ResourceMonitor_TypedConfig{
+						TypedConfig: protobuf.MustMarshalAny(
+							&envoy_fixed_heap_v3.FixedHeapConfig{
+								MaxHeapSizeBytes: c.MaximumHeapSizeBytes,
+							}),
+					},
+				},
+			},
+			Actions: []*envoy_config_overload_v3.OverloadAction{
+				{
+					Name: "envoy.overload_actions.shrink_heap",
+					Triggers: []*envoy_config_overload_v3.Trigger{
+						{
+							Name: "envoy.resource_monitors.fixed_heap",
+							TriggerOneof: &envoy_config_overload_v3.Trigger_Threshold{
+								Threshold: &envoy_config_overload_v3.ThresholdTrigger{
+									Value: 0.95,
```

Comment:
I could see folks wanting to tune these thresholds via flag, but I'm fine leaving statically defined for now since we can always add flags later if/when needed.

Reply:
True, these could be exposed in future, though I guess it means few more command line switches again...